src/core/Pagination/PageInput/PageInput.tsx

```diff
@@ -0,0 +1,259 @@
+import React, { ChangeEvent, Component, ReactNode } from 'react';
+import { default as styled } from 'styled-components';
+import classnames from 'classnames';
+import { SuomifiThemeProp, SuomifiThemeConsumer } from '../../theme';
+import { getConditionalAriaProp } from '../../../utils/aria';
+import {
+  HtmlInput,
+  HtmlInputProps,
+  HtmlSpan,
+  HtmlDiv,
+  HtmlButton,
+} from '../../../reset';
+import { VisuallyHidden } from '../../VisuallyHidden/VisuallyHidden';
+import { StatusText } from '../../Form/StatusText/StatusText';
+import { Label } from '../../Form/Label/Label';
+import { Icon } from '../../Icon/Icon';
+import { InputStatus, StatusTextCommonProps } from '../../Form/types';
+import { baseStyles } from './PageInput.baseStyles';
+
+export type PageInputValue = number | string | undefined;
+
+type PageInputStatus = Exclude<InputStatus, 'success'>;
+
+export interface PageInputProps
+  extends StatusTextCommonProps,
+    Omit<
+      HtmlInputProps,
+      | 'type'
+      | 'disabled'
+      | 'onChange'
+      | 'children'
+      | 'onClick'
+      | 'value'
+      | 'defaultValue'
+    > {
+  /** Label text */
+  labelText: ReactNode;
+  /** Placeholder text for input. */
+  visualPlaceholder?: string;
+  /** Page button label for screen readers */
+  pageInputButtonLabel: string;
+  /** Callback for search button click */
+  onPageChange?: (value: PageInputValue) => void;
+  /** Maximum value */
+  maxValue: number;
+  /** Callback for search button click */
```

Comment:
Should terminology here (onPageChange & invalidValueErrorText descriptions) be search button => page input button?